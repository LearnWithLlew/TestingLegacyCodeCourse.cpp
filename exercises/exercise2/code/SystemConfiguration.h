#pragma once

#include <string>
#include <sstream>

int rpm = 4000;
double dt = 0.01;
double x = 0;
double y = 0;
double z = 0;
double x_mph = 1;
double y_mph = 2;
double z_mph = 3;
bool gravity = true;
double y_torque = 3.2;

void initialiseSystem()
{
    if ( gravity )
    {
        double dx = x_mph / ( 60 * 60 * rpm );
        double dy = y_mph / ( 60 * 60 * rpm );
        for ( int i = 1; i <= rpm / 60; ++i )
        {
            x += dt * dx;
            y += dt * dy;
            z += dt * z_mph / ( 60 * 60 * rpm );
        }
    }
    else
    {
        double dx = x_mph / ( 60 * 60 * rpm );
        double dy = y_mph / ( 60 * 60 * rpm );
        for ( int i = 1; i <= rpm / 60; ++i )
        {
            x += dt * dx;
            y += dt * dy;
        }
    }
}
