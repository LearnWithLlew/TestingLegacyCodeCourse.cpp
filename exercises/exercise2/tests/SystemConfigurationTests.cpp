#include <catch2/catch.hpp>
#include "ApprovalTests.hpp"

#include "SystemConfiguration.h"

#include <iostream>

using namespace ApprovalTests;

TEST_CASE( "Test System Configuration 1" )
{
}

TEST_CASE( "generate code" )
{
}
