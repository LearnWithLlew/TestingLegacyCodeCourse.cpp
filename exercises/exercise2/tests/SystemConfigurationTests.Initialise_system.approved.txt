Initial state
rpm = 4000
dt = 0.01
x = 0
y = 0
z = 0
x_mph = 1
y_mph = 2
z_mph = 3
gravity = 0
y_torque = 3.2

State after initialization
rpm = 4000
dt = 0.01
x = 4.58333e-08
y = 9.16667e-08
z = 0
x_mph = 1
y_mph = 2
z_mph = 3
gravity = 0
y_torque = 3.2

